### PR TITLE
Update doc links to point to OpenSearch

### LINKED ISSUE #27: Update links to documentation
Current doc links are referencing the Open Distro for Elasticsearch documentation. Change to use the new https://docs-beta.opensearch.org/ home for docs.

### PR DESCRIPTION
**Pull request by ohltyler**
Signed-off-by: Tyler Ohlsen <ohltyler@amazon.com>

### Description

Updates the documentation links to point to the new home for docs: https://docs-beta.opensearch.org/

Refactored the base doc link into a constant to make it easier to change in the future (e.g., going from `docs-beta.opensearch` -> `docs.opensearch`). 

Sanity tested on the UI to make sure they work.

### Issues Resolved

Resolves #27 

### Check List
- [x] Commits are signed per the DCO using `--signoff`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by ylwu-amzn** [comment]
Searched "opendistro.github" in the branch `doc-links` and found this one https://github.com/ohltyler/anomaly-detection-dashboards-plugin-1/blob/doc-links/public/pages/DetectorConfig/containers/__tests__/__snapshots__/DetectorConfig.test.tsx.snap#L396

**Comment by ohltyler** [comment]
> Searched "opendistro.github" in the branch `doc-links` and found this one https://github.com/ohltyler/anomaly-detection-dashboards-plugin-1/blob/doc-links/public/pages/DetectorConfig/containers/__tests__/__snapshots__/DetectorConfig.test.tsx.snap#L396

Good catch. `yarn test:jest -u` didn't update this snapshot since it is related to a skipped test. Manually deleted the snapshot.

**Review by ylwu-amzn — APPROVED** [review]
LGTM. Thanks for the change!

File: README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [![Unit tests](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/workflows/Unit%20tests%20workflow/badge.svg)](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/actions?query=workflow%3A%22Unit+tests+workflow%22)
 [![Integration tests](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/workflows/E2E%20tests%20workflow/badge.svg)](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/actions?query=workflow%3A%22E2E+tests+workflow%22)
 [![codecov](https://codecov.io/gh/opensearch-project/anomaly-detection-dashboards-plugin/branch/main/graph/badge.svg)](https://codecov.io/gh/opensearch-project/anomaly-detection-dashboards-plugin)
-[![Documentation](https://img.shields.io/badge/doc-reference-blue)](https://opendistro.github.io/for-elasticsearch-docs/docs/ad/)
+[![Documentation](https://img.shields.io/badge/doc-reference-blue)](https://docs-beta.opensearch.org/docs/ad/)
```
Comment (spbjss) [inline README.md:4]: Would we change the "beta" to "release" or something else later?

Comment (ohltyler) [reply README.md:4]: Most likely. That's why I refactored it into a constant (see second line of PR description).

**Review by spbjss — APPROVED** [review]
looks good to me.

**Pull request merged**